napari_allencell_annotator/view/annotator_view.py

```diff
@@ -50,75 +99,189 @@ def __init__(self, viewer: napari.Viewer, annot_data: Dict[str, Dict[str, str]]
+        2px;} """
+        self.scroll.setStyleSheet(self.scroll.styleSheet() + style)
         self.layout.addWidget(self.scroll, 1, 0, 10, 4)
 
-        self._render_gui(annot_data)
+        self.num_images: int = None
+        self.curr_index: int = None
+        self.contr = contr
 
-        self.setLayout(self.layout)
-        self.viewer = viewer
+        # Add widget visible in ADD mode
+        self.add_widget = QWidget()
+        add_layout = QHBoxLayout()
+        self.create_btn = QPushButton("Create New Annotations")
+        self.create_btn.setEnabled(True)
+        self.import_btn = QPushButton(
+            "Import Existing Annotations (.csv or .json)"
+        )
+        self.import_btn.setEnabled(True)
+
+        add_layout.addWidget(self.create_btn, stretch=2)
+        add_layout.addWidget(self.import_btn, stretch=2)
+        self.add_widget.setLayout(add_layout)
+        self.layout.addWidget(self.add_widget, 12, 0, 1, 4)
+        self.add_widget.hide()
+
+        # view widget visible in VIEW mode
+        self.view_widget = QWidget()
+        view_layout = QHBoxLayout()
+        self.cancel_btn = QPushButton("Cancel")
+        self.cancel_btn.setEnabled(False)
+        self.start_btn = QPushButton("Start Annotating")
+        self.file_input = FileInput(
+            mode=FileInputMode.CSV, placeholder_text="Start Annotating"
+        )
+        self.file_input.toggle(False)
+        self.start_btn.setEnabled(True)
 
-        self.show()
+        view_layout.addWidget(self.cancel_btn, stretch=1)
+        view_layout.addWidget(self.start_btn, stretch=3)
+        self.view_widget.setLayout(view_layout)
+        self.layout.addWidget(self.view_widget, 12, 0, 1, 4)
+        self.view_widget.hide()
+
+        # annot widget visible in ANNOTATE mode
+        self.annot_widget = QWidget()
+        annot_layout = QGridLayout()
+        self.prev_btn = QPushButton("< Previous")
+        self.next_btn = QPushButton("Next >")
+        self.next_btn.setEnabled(True)
+        self.progress_bar = QLabel()
+        annot_layout.addWidget(self.progress_bar, 0, 1, 1, 2)
+        annot_layout.addWidget(self.prev_btn, 1, 0, 1, 2)
+        annot_layout.addWidget(self.next_btn, 1, 2, 1, 2)
+        self.annot_widget.setLayout(annot_layout)
+        self.layout.addWidget(self.annot_widget, 12, 0, 2, 4)
+        self.view_widget.hide()
+
+        self._display_mode()
+        self.annotation_item_widgets: List[QWidget] = []
+        self.annots_order: List[str] = []
+        self.default_vals: List[str] = []
+        self.setLayout(self.layout)
+        self.viewer: napari.Viewer = viewer
 
     @property
     def mode(self) -> AnnotatorViewMode:
         return self._mode
 
-    def _render_gui(self, annot_data: Dict[str, Dict[str, str]]):
+    def set_mode(self, mode: AnnotatorViewMode):
+        self._mode = mode
+        self._display_mode()
+
+    def set_num_images(self, num: int):
```

Comment:
if you never actually use `self.num_images` while it `= None`, you can just make this function a one liner-
`self.num_images = num`. if you do I'd keep the check for None